src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +192,100 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
         return factorsByVarConfig;
     }
 
-    public List<SensitivityValue> analyse(Network network, List<SensitivityFactor> factors, OpenLoadFlowParameters lfParametersExt,
+    /**
+     * If the slack is distributed, we need to create a factor for every generator
+     * @param network
+     * @param lfNetwork
+     * @param oldFactors
+     * @return
+     */
+    private List<SensitivityFactor> getDistributedFactors(Network network, LfNetwork lfNetwork,
```

Comment:
This method uses network to get generators. But lfNetwork may not contain all these generators (because it is a connected component of network), so I would love to use only lfNetwork, but I do not know how to access generators from LfNetwork